usd/src/sdf_parser/Model.cc

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "sdf/usd/sdf_parser/Model.hh"
+
+#include <string>
+#include <unordered_map>
+
+#include <ignition/math/Pose3.hh>
+#include <ignition/math/Vector3.hh>
+
+// TODO(ahcorde) this is to remove deprecated "warnings" in usd, these warnings
+// are reported using #pragma message so normal diagnostic flags cannot remove
+// them. This workaround requires this block to be used whenever usd is
+// included.
+#pragma push_macro ("__DEPRECATED")
+#undef __DEPRECATED
+#include <pxr/usd/sdf/path.h>
+#include <pxr/usd/usd/stage.h>
+#include <pxr/usd/usdGeom/xform.h>
+#pragma pop_macro ("__DEPRECATED")
+
+#include "sdf/Model.hh"
+#include "sdf/usd/sdf_parser/Link.hh"
+#include "../UsdUtils.hh"
+
+namespace sdf
+{
+// Inline bracke to help doxygen filtering.
+inline namespace SDF_VERSION_NAMESPACE {
+//
+namespace usd
+{
+  UsdErrors ParseSdfModel(const sdf::Model &_model, pxr::UsdStageRefPtr &_stage,
+      const std::string &_path, const pxr::SdfPath &/*_worldPath*/)
+  {
+    UsdErrors errors;
+
+    if (_model.ModelCount())
+    {
+      errors.push_back(UsdError(
+            sdf::Error(sdf::ErrorCode::NESTED_MODELS_UNSUPPORTED,
+              "Nested models currently aren't supported.")));
+      return errors;
+    }
+
+    const pxr::SdfPath sdfModelPath(_path);
+    auto usdModelXform = pxr::UsdGeomXform::Define(_stage, sdfModelPath);
+    // since USD does not have a plane yet, planes are being represented as a
+    // wide, thin box. The plane/box pose needs to be offset according to the
+    // plane thickness to ensure that the top of the plane is at the correct
+    // height. This pose offset workaround will no longer be needed when a
+    // pxr::USDGeomPlane class is created:
+    // https://graphics.pixar.com/usd/release/wp_rigid_body_physics.html#plane-shapes
+    if (usd::IsPlane(_model))
+    {
+      static const double kPlaneThickness = 0.25;
+      ignition::math::Vector3d planePosition(
+          _model.RawPose().X(),
+          _model.RawPose().Y(),
+          _model.RawPose().Z() - (0.5 * kPlaneThickness));
+      const auto poseErrors = usd::SetPose(
```

Comment:
Not related to this PR, but I see that `usd::SetPose` is declared with `IGNITION_SDFORMAT_USD_VISIBLE`, but it is in a header file under `usd/src/UsdUtils.hh` so it won't be avilable as a public api. Should we move that to `usd/include` so that other libraries like `ignition-omni` can make use of it?

Reply:
in that case we should remove the visibility macro

Reply:
I created a PR that removes this macro if you wouldn't mind taking a look: #857